ironfish/src/blockchain/nullifierSet/nullifierSet.ts

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { Assert } from '../../assert'
+import { Transaction } from '../../primitives'
+import { Nullifier } from '../../primitives/nullifier'
+import {
+  BUFFER_ENCODING,
+  DatabaseSchema,
+  IDatabase,
+  IDatabaseStore,
+  IDatabaseTransaction,
+  StringEncoding,
+  U32_ENCODING,
+} from '../../storage'
+import { NullifierInfo, NullifierLocationEncoding } from './encoding'
+
+interface CounterEntry<T extends string> extends DatabaseSchema {
+  key: T
+  value: number
+}
+
+interface NullifiersSchema extends DatabaseSchema {
+  key: Nullifier
+  value: NullifierInfo
+}
+
+export class NullifierSet {
+  readonly db: IDatabase
+
+  // Keep track of number of nullifiers in the set
+  private readonly counter: IDatabaseStore<CounterEntry<'Size'>>
+
+  // Nullifier -> BlockHash, TransactionHash
```

Comment:
Does this need to be updated to something like `Nullifier -> BlockHash, position`? Same with the below. I haven't looked to see what this is in other places

Reply:
Yea it should be `Nullifier -> TransactionHash, position`. Updated